eta/core/c3d.py

```diff
@@ -0,0 +1,384 @@
+            d, "num_frames_per_clip", default=16)
+        self.stride = self.parse_number(d, "stride", default=8)
+
+
+class C3D(object):
+    '''C3D network structure in TensorFlow.'''
+
+    def __init__(self, config=None, sess=None, clips=None):
+        '''Builds a new C3D network
+
+        Args:
+            config: an optional C3DConfig instance. If omitted, the default
+                ETA configuration will be used
+            sess: an optional tf.Session to use. If none is provided, a new
+                tf.Session instance is created, and you are responsible for
+                scalling the close() method of this class when you are done
+                computing
+            clips: an optional tf.placeholder of size [XXXX,
+            num_frames_per_clip,embedding_frame_size, embedding_frame_size, 3]
+            to use. By default, a placeholder of size [None,
+            num_frames_per_clip, embedding_frame_size, embedding_frame_size, 3]
+            is used so you can evaluate any number of images at once
+        '''
+        self.config = config or C3DConfig.default()
+        self.sess = sess or tf.Session()
+        self.clips = clips or tf.placeholder(
+            tf.float32, [None,
+            self.config.num_frames_per_clip,self.config.embedding_frame_size,
+            self.config.embedding_frame_size, 3])
+        self.build_c3d(self.clips, self.config.dropout)
+        self._load_model(self.config.model)
+
+    def _variable_with_weight_decay(self, name, shape, stddev, wd):
+        var = tf.get_variable(
+            name, shape,
+            initializer=tf.truncated_normal_initializer(stddev=stddev))
+        if wd is not None:
+            weight_decay = tf.nn.l2_loss(var) * wd
+            tf.add_to_collection('losses', weight_decay)
+        return var
+
+    def _conv3d(self, name, l_input, w, b):
+        return tf.nn.bias_add(
+            tf.nn.conv3d(
+                l_input, w, strides=[1, 1, 1, 1, 1], padding='SAME'), b)
+
+    def _max_pool(self, name, l_input, k):
+        return tf.nn.max_pool3d(
+            l_input, ksize=[1, k, 2, 2, 1],
+            strides=[1, k, 2, 2, 1], padding='SAME', name=name)
+
+    def evaluate(self, clips, layer=None):
+        '''Feed-forward evaluation through the net.
+
+        Args:
+            imgs: an array of size [XXXX, 112, 112, 3] containing image(s) to
+                feed into the network
+            layer: an optional layer whose output to return. By default, the
+                output softmax layer (i.e., the class probabilities) is
+                returned
+        '''
+        if layer is None:
+            layer = self.out
+        return self.sess.run(layer, feed_dict={self.clips: [clips]})[0]
+
+    def close(self):
+        '''Closes the TensorFlow session and frees up the network.'''
+        self.sess.close()
+        self.sess = None
+
+    def build_c3d(self, _X, _dropout):
+        with tf.variable_scope('var_name') as var_scope:
```

Comment:
this is still not quite what I requested: I wanted you to use `tf.name_scope()` to give a named scope to each layer. This command `tf.variable_scope('var_name')` is non-sensical.

See `eta.core.vgg16` to see how I'd like to see the model constructed with named layers.